pallets/collator-assignment/src/lib.rs

```diff
@@ -161,6 +162,10 @@ pub mod pallet {
             // We get the containerChains that we will have at the target session
             let mut container_chain_ids =
                 T::ContainerChains::session_container_chains(target_session_index);
+            // Remove the containerChains that do not have enough credits for block production
+            T::RemoveParaIdsWithNoCredits::remove_para_ids_with_no_credits(
```

Comment:
this requires new reads, hence the weight should be corrected

Reply:
Actually now the benchmarks are broken because this function removes all the registered chains...

Reply:
so we should add a function to `RemoveParaIdsWithNoCredits` with runtime-benchmarks, which sets the credit

Reply:
But I believe the value used for the weight hint is wrong, it uses the number of parachains after filtering out the ones that do not have enough credit:

```rust
            let assigned_collators = Self::assign_collators(session_index, random_seed, collators);
            let num_parachains = assigned_collators.next_assignment.container_chains.len();
            frame_system::Pallet::<T>::register_extra_weight_unchecked(
                T::WeightInfo::new_session(num_collators as u32, num_parachains as u32),
                DispatchClass::Mandatory,
            );
```

Instead it should use the initial number of parachains, I guess I will add a new return param to the `assign_collators`  function.

Reply:
(done)